crates/services/gas_price_service/src/v0/tests.rs

```diff
@@ -183,64 +211,152 @@ async fn next__new_l2_block_saves_old_metadata() {
 
     // when
     let service = ServiceRunner::new(service);
-    let shared = service.shared.clone();
-    let start = shared.next_gas_price();
 
     service.start_and_await().await.unwrap();
     l2_block_sender.send(l2_block).await.unwrap();
     tokio::time::sleep(Duration::from_millis(10)).await;
 
     // then
-    let new = shared.next_gas_price();
-    assert_ne!(start, new);
+    assert!(metadata_inner.lock().unwrap().is_some());
+}
+
+#[derive(Clone)]
+struct FakeSettings;
+
+impl GasPriceSettingsProvider for FakeSettings {
+    fn settings(
+        &self,
+        _param_version: &ConsensusParametersVersion,
+    ) -> GasPriceResult<GasPriceSettings> {
+        todo!()
+    }
+}
+
+#[derive(Clone)]
+struct FakeGasPriceDb;
+
+// GasPriceData + Modifiable + KeyValueInspect<Column = GasPriceColumn>
+impl GasPriceData for FakeGasPriceDb {
+    fn latest_height(&self) -> Option<BlockHeight> {
+        todo!()
```

Comment:
Dangling todo. Shuld we remove this function or when is it to be implemented?

Reply:
It's fine. It just doesn't get called in the scope of any of these tests, but we're still required to pass something that implements that trait.

Reply:
I can change to `unimplemented` if that's better.

Reply:
Yeah then `unimplemented` conveys the intention better.